server.js

```diff
@@ -1,33 +1,106 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import netflixData from './data/netflix-titles.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
+// PORT SETUP
 // Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// overridden when starting the server. For example: PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// MIDDLEWARES 
+// To enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// ROOT ENDPOINT
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('API for Netflix shows, available endpoints: /shows -> /:id, /movies, /tv-shows, ?title=X for search and ?page=X for pagination')
+})
+
+// GET ALL DATA WITH PAGINATION OR FILTERED WITH TITLE IN QUERY
+// req = request-object, res = result-object
+app.get('/shows', (req, res) => {
+  const page = req.query.page
+  const titleSearchString = req.query.title
+  // const typeSearchString = req.query.type
+  let filteredShows = netflixData
+  const PER_PAGE = 20 // Sets 20 shows per page
+
+  if (titleSearchString) {
+    filteredShows = filteredShows.filter((item) => {
+      const itemTitle = item.title.toString().toLowerCase() // toString since titles might be number
+      return itemTitle.includes(titleSearchString.toLowerCase())
+    })
+  }
+
+  // if (typeSearchString) {
+  //   filteredShows = filteredShows.filter((item) => item.type.toLowerCase().replace(' ', '-') === typeSearchString.toLowerCase())
+  // }
+
+  if (page) {
+    const startIndex = PER_PAGE * +page
+    // StartIndex and the upcoming shows updates when increasing page number in endpoint
+    filteredShows = filteredShows.slice(startIndex, startIndex + PER_PAGE)
+  }
```

Comment:
I think it'd be nice to have this pagination as a default - returning so many responses from the API as a default is pretty heavy, and no frontend is going to be ready for that many results all at once - better to break them up!